vst2_windows.go

```diff
@@ -30,34 +29,32 @@ type handle struct {
 func init() {
 	envVstPath := os.Getenv("VST_PATH")
 	if len(envVstPath) > 0 {
-		scanPaths = append(paths, envVstPath)
+		scanPaths = append(scanPaths, envVstPath)
 	}
 }
 
 // open loads the plugin entry point into memory. It's DLL in windows.
 func open(path string) (effectMain, handle, error) {
 	//Load plugin by path
-	dll, err := syscall.LoadDLL(l.Path)
+	dll, err := syscall.LoadDLL(path)
 	if err != nil {
-		return nil, handle{}, fmt.Errorf("failed to load VST from '%s': %v\n", l.Path, err)
+		return nil, handle{}, fmt.Errorf("failed to load VST from '%s': %v\n", path, err)
 	}
-	l.library = unsafe.Pointer(dll)
-	l.Name = strings.TrimSuffix(filepath.Base(dll.Name), filepath.Ext(dll.Name))
 
 	//Get pointer to plugin's Main function
 	m, err := syscall.GetProcAddress(dll.Handle, main)
 	if err != nil {
-		l.Close()
-		return nil, handle{}, fmt.Errorf("failed to get entry point for plugin'%s': %v\n", l.Path, err)
+		_ = dll.Release()
+		return nil, handle{}, fmt.Errorf("failed to get entry point for plugin'%s': %v\n", path, err)
 	}
-	return effectMain(m), handle{dll: dll}, nil
+
+	return effectMain(unsafe.Pointer(m)), handle{dll: dll}, nil
```

Comment:
You are right about proper handling for release error. I believer that the following snippet will work: 

```go
m, err := syscall.GetProcAddress(dll.Handle, main)
if err == nil {
    return effectMain(unsafe.Pointer(m)), handle{dll: dll}, nil
}
err = fmt.Errorf("failed to get entry point for plugin '%s': %w\n", path, err)
if errRelease := dll.Release(); errRelease != nil {
    return nil, handle{}, fmt.Errorf("failed to release DLL '%s': %w after: %v", path, errRelease, err)
}
return nil, handle{}, err
```

What do you think?

Reply:
thank you very much.
i added your snippet and some build constraints.
However it turned out, that the example was failing on windows due to a null pointer exception in c (the error from the first post of this pr).
I can not fix the error (it has something to do with `vst2.(*Plugin).ProcessDouble`, but when i add a second channel to the input data there is no null pointer anymore (the same error occurs, if i remove a channel from `vst_test.go`).
Furthermore i have no darwin machine to test the example `// Output:` for the `example_vst2_darwin_test.go`.
Please run `go test -v example_vst2_darwin_test.go` on a darwin machine and fix the `// Output:` if needed - perhaps the c null pointer occurs on darwin as well?

Reply:
i just fixed the drawin output according to the travis build fail... seems to work on darwin with only one channel.